pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts

```diff
@@ -0,0 +1,802 @@
+
+        await advanceTime(delay);
+        await authorizer.execute(id);
+
+        expect(await authorizer.hasPermission(ACTION_3, user, WHERE_1)).to.be.equal(false);
+        expect(await authorizer.hasPermission(ACTION_2, user, WHERE_2)).to.be.equal(false);
+      });
+
+      it('emits an event', async () => {
+        const receipt = await authorizer.instance
+          .connect(getSender())
+          .scheduleRevokePermission(ACTION_1, user.address, WHERE_1, []);
+
+        expectEvent.inReceipt(await receipt.wait(), 'RevokePermissionScheduled', {
+          actionId: ACTION_1,
+          account: user.address,
+          where: WHERE_1,
+        });
+      });
+    }
+
+    context('when the sender is root', () => {
+      itScheduleRevokePermissionCorrectly(() => root);
+    });
+
+    context('when the sender is revoker', () => {
+      sharedBeforeEach('makes a revoker', async () => {
+        await authorizer.addRevoker(revoker, EVERYWHERE, { from: root });
+      });
+      itScheduleRevokePermissionCorrectly(() => revoker);
+    });
+  });
+
+  describe('renouncePermission', () => {
+    context('when the sender does not have the permission', () => {
+      context('when renouncing the permission for a specific contract', () => {
+        it('ignores the request and still cannot perform the requested action everywhere', async () => {
+          await expect(authorizer.renouncePermission(ACTION_1, WHERE_1, { from: user })).not.to.be.reverted;
+
+          expect(await authorizer.canPerform(ACTION_1, user, EVERYWHERE)).to.be.false;
+        });
+
+        it('ignores the request and still cannot perform the requested action in any specific contract', async () => {
+          await expect(authorizer.renouncePermission(ACTION_1, WHERE_1, { from: user })).not.to.be.reverted;
+
+          expect(await authorizer.canPerform(ACTION_1, user, NOT_WHERE)).to.be.false;
+          expect(await authorizer.canPerform(ACTION_1, user, WHERE_2)).to.be.false;
+        });
+      });
+
+      context('when renouncing the permission for everywhere', () => {
+        it('ignores the request and still cannot perform the requested action everywhere', async () => {
+          await expect(authorizer.renouncePermissionGlobally(ACTION_1, { from: user })).not.to.be.reverted;
+
+          expect(await authorizer.canPerform(ACTION_1, user, EVERYWHERE)).to.be.false;
+        });
+
+        it('ignores the request and still cannot perform the requested action in any specific contract', async () => {
+          await expect(authorizer.renouncePermissionGlobally(ACTION_1, { from: user })).not.to.be.reverted;
+
+          expect(await authorizer.canPerform(ACTION_1, user, NOT_WHERE)).to.be.false;
+          expect(await authorizer.canPerform(ACTION_1, user, WHERE_2)).to.be.false;
+        });
+      });
+    });
+
+    context('when the user has the permission for a specific contract', () => {
```

Comment:
I'd add a test where `revoke` has a delay (so we can only do `scheduleRevoke`), but `renounce` still works immediately, with no delay.